crates/red_knot_python_semantic/src/program.rs

```diff
@@ -142,11 +143,21 @@ pub enum PythonPath {
     /// `/opt/homebrew/lib/python3.X/site-packages`.
     ///
     /// [`sys.prefix`]: https://docs.python.org/3/library/sys.html#sys.prefix
-    SysPrefix(SystemPathBuf),
+    SysPrefix(SystemPathBuf, SysPrefixPathOrigin),
```

Comment:
Same here. Should this store a `SysPrefixPath` instead of the two individual fields

Reply:
`SysPrefixPath` represents a _validated_ path to `sys.prefix`, but at this stage the path hasn't yet been validated (and I don't think we want to do the validation this early; it's better if we do it in `SearchPaths::from_settings()`)

Reply:
I see. It's somewhat annoying that we have to pass through all those arguments and I'm not sure if it's necessary to restrict `SysPrefixPath` to *valid paths* only. But changing this constraint seems out of the scope for this PR

Reply:
yeah, agreed, we could consider a redesign where the validation is moved out of `SysPrefixPath::new()`. But I also agree that it's out of scope here.